### ISSUE TITLE
Problems Migrating Database from 8.17.2 to 9.3.0

### ISSUE THREAD

**Issue by hfloyd** [umbraco/Umbraco-CMS#11976]
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

8.17.2 to 9.3.0

### Bug summary

The error received is: 

Error during installation
The database failed to upgrade. ERROR: The database configuration failed with the following message: Invalid column name 'userOrMemberKey'. Please check log file for additional information (can be found in '~/umbraco/Logs')

See the log for full details (logs can typically be found in the umbraco\Logs folder).

**Is it safe to just delete these non-umbraco tables/columns?**

### Specifics

This is an upgrade from a version 7.4.4 site all the way to 9.3.0. The DB does include some old third-party tables. I was able to complete the upgrade through 8.17.2 (able to log into the back-office, see the content, etc.)

The Log info just prior to the failure includes:

```
`The database schema validation produced the following summary: 

"The following tables were found in the database, but are not in the current schema: SEOChecker_URLHistory,SEOChecker_ValidationIssues,UCErrorLog,UFRecordDataBit,UFRecordDataDateTime,UFRecordDataInteger,UFRecordDataLongString,UFRecordDataString,UFRecordFields,UFRecords,UFUserFormSecurity,UFUserSecurity,BulkmanagerState,CMSImportMediaRelation,CMSImportRelation,CMSImportScheduledItems,CMSImportScheduledTask,CMSImportState,SEOChecker_ConfigurationIssues,SEOChecker_Notifications,SEOChecker_PageNotFound,SEOChecker_Queue,SEOChecker_Robots,SEOChecker_ScheduledValidation,umbracoExternalLoginToken,umbracoTwoFactorLogin,umbracoLogViewerQuery,umbracoContentVersionCleanupPolicy,umbracoUserGroup2Node,umbracoCreatedPackageSchema 

The following columns were found in the database, but are not in the current schema: SEOChecker_ValidationIssues,DataTypeId,UFRecordDataBit,Key,UFRecordDataDateTime,Key,UFRecordDataInteger,Key,UFRecordDataLongString,Key,UFRecordFields,Key,UFRecordFields,FieldId,UFRecordDataString,Key,UFRecords,CurrentPage,UFRecords,Form,UFRecords,UniqueId,UFUserFormSecurity,Form,BulkmanagerState,StateId,CMSImportScheduledTask,ScheduleGUID,CMSImportScheduledTask,ImportStateGUID,CMSImportState,UniqueIdentifier,CMSImportState,Parent,SEOChecker_ScheduledValidation,Id,SEOChecker_ScheduledValidation,ExecuteHour,SEOChecker_ScheduledValidation,ExecuteMinute,SEOChecker_ScheduledValidation,ValidationRoot,SEOChecker_Notifications,Id,SEOChecker_Notifications,ExecuteHour,SEOChecker_Notifications,ExecuteMinute,SEOChecker_PageNotFound,NotFoundId,SEOChecker_PageNotFound,DocumentID,SEOChecker_PageNotFound,TimesAccessed,SEOChecker_Queue,DocumentID,SEOChecker_Queue,Priority,SEOChecker_Robots,NodeId,SEOChecker_ConfigurationIssues,IssueId,CMSImportScheduledTask,ScheduleId,CMSImportMediaRelation,Id,CMSImportMediaRelation,UmbracoMediaId,CMSImportMediaRelation,ByteSize,CMSImportRelation,Id,CMSImportRelation,UmbracoID,CMSImportScheduledItems,ScheduledItemId,CMSImportScheduledItems,ScheduleldTaskId,CMSImportScheduledTask,ExecuteHour,CMSImportScheduledTask,ExecuteMinute,CMSImportScheduledTask,ImportAsUser,CMSImportState,Id,BulkmanagerState,Id,BulkmanagerState,UserId,UFUserFormSecurity,SecurityType,umbracoExternalLogin,userId,UFUserFormSecurity,Id,UFRecords,UmbracoPageId,UFRecordDataString,Id,UFRecords,Id,UFRecordFields,Record,UFRecordDataInteger,Value,UFRecordDataLongString,Id,UFRecordDataDateTime,Id,UFRecordDataInteger,Id,UFRecordDataBit,Id,UCErrorLog,Id,SEOChecker_URLHistory,UrlHistoryId,SEOChecker_URLHistory,NodeId,SEOChecker_URLHistory,ParentId,SEOChecker_ValidationIssues,IssueId,SEOChecker_ValidationIssues,DocumentID,SEOChecker_ValidationIssues,CrawlDate,SEOChecker_ValidationIssues,ResolvedDate,CMSImportScheduledItems,ScheduledOn,CMSImportScheduledItems,ExecutedOn,CMSImportRelation,Updated,SEOChecker_Queue,QueuedDateTime,SEOChecker_PageNotFound,LastTimeAccessed,SEOChecker_Notifications,NextRun,SEOChecker_ConfigurationIssues,CrawlDate,SEOChecker_ConfigurationIssues,ResolvedDate,SEOChecker_Robots,RequestDateTime,SEOChecker_ScheduledValidation,NextRun,SEOChecker_ScheduledValidation,LastTimeExecuted,UCErrorLog,Date,UFRecordDataDateTime,Value,UFRecords,Created,UFRecords,Updated,UFRecords,RecordData,UFRecordDataLongString,Value,SEOChecker_PageNotFound,Ignore,SEOChecker_ConfigurationIssues,Ignore,SEOChecker_PageNotFound,WildCard,SEOChecker_ScheduledValidation,IncludeChildren,BulkmanagerState,BindOnLoad,umbracoServer,isMaster,CMSImportScheduledItems,InProgress,UFUserSecurity,ManageDataSources,UFUserSecurity,ManagePreValueSources,UFUserSecurity,ManageWorkflows,UFUserSecurity,ManageForms,UFUserFormSecurity,HasAccess,UFUserFormSecurity,AllowInEditor,UCErrorLog,Read,UFRecordDataBit,Value,SEOChecker_ValidationIssues,Ignore,UCErrorLog,Thread,UCErrorLog,Level,UCErrorLog,Logger,UCErrorLog,Message,UCErrorLog,Exception,UCErrorLog,ReadBy,SEOChecker_URLHistory,UrlPart,SEOChecker_ValidationIssues,ErrorCode,SEOChecker_ValidationIssues,ValidatorAlias,SEOChecker_ValidationIssues,ErrorDescription,SEOChecker_ValidationIssues,Error,SEOChecker_ValidationIssues,Template,SEOChecker_ValidationIssues,Category,SEOChecker_ValidationIssues,ValidationType,SEOChecker_ValidationIssues,ErrorType,SEOChecker_ValidationIssues,PropertyAlias,UFUserFormSecurity,User,UFRecords,State,UFRecordFields,Alias,UFRecordFields,DataType,UFRecordDataString,Value,UFRecords,IP,UFRecords,MemberKey,UFUserSecurity,User,CMSImportRelation,DataSourceKey,CMSImportRelation,ImportProvider,CMSImportMediaRelation,SourceUrl,CMSImportRelation,CustomId,CMSImportState,ImportProvider,CMSImportState,Name,CMSImportState,ImportState,CMSImportScheduledTask,ScheduledTaskName,CMSImportScheduledTask,NotifyEmailAddress,CMSImportScheduledTask,ExecuteEvery,CMSImportScheduledTask,ExecuteDays,BulkmanagerState,state,BulkmanagerState,Name,BulkmanagerState,ProviderId,SEOChecker_PageNotFound,Url,SEOChecker_Notifications,NotificationUser,SEOChecker_Notifications,ExecuteEvery,SEOChecker_Notifications,ExecuteDays,SEOChecker_PageNotFound,RedirectUrl,SEOChecker_PageNotFound,Referer,SEOChecker_PageNotFound,ContentType,SEOChecker_PageNotFound,Domain,SEOChecker_PageNotFound,QueryString,SEOChecker_Robots,UserAgent,SEOChecker_ScheduledValidation,TaskName,SEOChecker_ScheduledValidation,NotificationEmailAddress,SEOChecker_ScheduledValidation,ExecuteEvery,SEOChecker_ScheduledValidation,ExecuteDays,SEOChecker_ConfigurationIssues,ErrorCode,SEOChecker_ConfigurationIssues,ValidatorAlias,SEOChecker_ConfigurationIssues,ErrorDescription,SEOChecker_ConfigurationIssues,Error,SEOChecker_ConfigurationIssues,ErrorType,SEOChecker_ConfigurationIssues,Category,umbracoContentVersion,preventCleanup,cmsMember,passwordConfig,cmsMember,securityStampToken,cmsMember,emailConfirmedDate,umbracoServer,isSchedulingPublisher,umbracoExternalLogin,userOrMemberKey,umbracoExternalLoginToken,id,umbracoExternalLoginToken,externalLoginId,umbracoExternalLoginToken,name,umbracoExternalLoginToken,value,umbracoExternalLoginToken,createDate,umbracoTwoFactorLogin,id,umbracoTwoFactorLogin,userOrMemberKey,umbracoTwoFactorLogin,providerName,umbracoTwoFactorLogin,secret,umbracoLogViewerQuery,id,umbracoLogViewerQuery,name,umbracoLogViewerQuery,query,umbracoContentVersionCleanupPolicy,contentTypeId,umbracoContentVersionCleanupPolicy,preventCleanup,umbracoContentVersionCleanupPolicy,keepAllVersionsNewerThanDays,umbracoContentVersionCleanupPolicy,keepLatestVersionPerDayForDays,umbracoContentVersionCleanupPolicy,updated,umbracoUserGroup2Node,userGroupId,umbracoUserGroup2Node,nodeId,umbracoCreatedPackageSchema,id,umbracoCreatedPackageSchema,name,umbracoCreatedPackageSchema,value,umbracoCreatedPackageSchema,updateDate,umbracoCreatedPackageSchema,packageId 

The following constraints (Primary Keys, Foreign Keys and Indexes) were found in the database, but are not in the current schema: FK_UFRecordDataBit_UFRecordFields_Key,FK_UFRecordDataDateTime_UFRecordFields_Key,FK_UFRecordDataInteger_UFRecordFields_Key,FK_UFRecordDataLongString_UFRecordFields_Key,FK_UFRecordDataString_UFRecordFields_Key,FK_umbracoExternalLoginToken_umbracoExternalLogin_id,FK_umbracoContentVersionCleanupPolicy_cmsContentType_nodeId,FK_umbracoUserGroup2Node_umbracoUserGroup_id,FK_umbracoUserGroup2Node_umbracoNode_id,PK_[SEOChecker_ConfigurationIssues,PK__UCErrorL__3214EC0743483E52,PK_BulkmanagerState,PK_CMSImportMediaRelation,PK_CMSImportRelation,PK_CMSImportScheduledItems,PK_CMSImportScheduledTask,PK_CMSImportState,PK_SEOChecker_Notifications,PK_SEOChecker_PageNotFound,PK_SEOChecker_Queue,PK_SEOChecker_ScheduledValidation,PK_SEOChecker_URLHistory,PK_SEOChecker_ValidationIssues,PK_UFRecordDataBit,PK_UFRecordDataDateTime,PK_UFRecordDataInteger,PK_UFRecordDataLongString,PK_UFRecordDataString,PK_UFRecordFields,PK_UFRecords,PK_umbracoExternalLoginToken,PK_umbracoTwoFactorLogin,PK_umbracoLogViewerQuery,PK_umbracoUserGroup2Node,PK_umbracoCreatedPackageSchema 

The following indexes were found in the database, but are not in the current schema: IX_databit_recordfield,IX_datadatetime_recordfield,IX_datainteger_recordfield,IX_datalongstring_recordfield,IX_datastring_recordfield,IX_cmsLanguageText_languageId,IX_umbracoExternalLogin_userOrMemberKey,IX_umbracoExternalLogin_LoginProvider,IX_umbracoExternalLogin_ProviderKey,IX_umbracoExternalLoginToken_Name,IX_umbracoTwoFactorLogin_userOrMemberKey,IX_umbracoTwoFactorLogin_ProviderName,IX_LogViewerQuery_name,IX_umbracoUserGroup2Node_nodeId,IX_umbracoCreatedPackageSchema_Name 

The following unknown constraints (Primary Keys, Foreign Keys and Indexes) were found in the database, but are not in the current schema: Id "`
```

### Steps to reproduce

Try migrating a v 7.4.4 site to v 9.3.0 ;-)

### Expected result / actual result

I was hoping the 9.3.0 migration could occur without issues, and any invalid/unneeded columns automatically removed

**Comment by hfloyd** [comment]
So, I thought perhaps I could locate this invalid column and remove it...  I used SSMS and ran "Tasks > Generate Scripts" (to get the whole schema written out to a text file), but a search through that file for `userOrMemberKey` doesn't turn up any matches. so, I'm a bit stumped about WHERE this column is located.

**Comment by marcemarc** [comment]
> So, I thought perhaps I could locate this invalid column and remove it... I used SSMS and ran "Tasks > Generate Scripts" (to get the whole schema written out to a text file), but a search through that file for `userOrMemberKey` doesn't turn up any matches. so, I'm a bit stumped about WHERE this column is located.

https://github.com/umbraco/Umbraco-CMS/blob/8eb3ef65f686619d09af478bd6fd563fc90580b2/src/Umbraco.Infrastructure/Migrations/Upgrade/V_9_3_0/UpdateExternalLoginToUseKeyInsteadOfId.cs looks like its a new column added for 9.3.0... But is issue logging in to trigger the migration? Wonder if it would work if you migrated to 9.2 first?

**Comment by hfloyd** [comment]
So, I went back to v8.17.2 and added a new user. I then attached the Db to v9.2, and "Authorized" the upgrade using that new user. It seemed to complete - the message said "Installation is complete! Get ready to be redirected to your new CMS." and then that same "HTTP Error 404.15 - Not Found - The request filtering module is configured to deny a request where the query string is too long." error was thrown. 

But, I figured I'd give the further update a go, and carried onw itht he 9.3.0 update. That one showed the "Installation is complete! Get ready to be redirected to your new CMS." message almost immediately, and actually DID then load the Umbraco back-office!

Phew! Thanks again @marcemarc !
